src/main/java/ui/UI.java

```diff
@@ -217,6 +221,16 @@ private void initPreApplicationState() {
 
         uiManager = new UIManager(this);
         status = new HTStatusBar(this);
+
+        updateProgressWindow = new UpdateProgressWindow();
+
+        if (!TestController.isTestMode()) {
+            updateManager = new UpdateManager(updateProgressWindow);
+        } else {
+            updateManager = new UpdateManagerStub();
```

Comment:
Something like this? Or should I abstract this out to a method somewhere (not sure which class should the instancing belong to)?